ir/function.cpp

```diff
@@ -299,6 +374,112 @@ void Function::unroll(unsigned k) {
   LoopAnalysis la(*this);
   ofstream out("a.gv");
   la.printDot(out);
+
+  auto &nodes = la.getHeaderNodes();
+  auto &roots = la.getRoots();
+
+  BasicBlock &sink = getBB("#sink");
+
+  vector<pair<const BasicBlock*, bool>> worklist;
+
+  for (auto &root : roots) {
+    worklist.emplace_back(root, false);
+  }
+
+  while (!worklist.empty()) {
+    auto &[bb, flag]  = worklist.back();
+    if (flag) {
```

Comment:
Let's use `if (!flag) { ... (the code that uses getChildren()); continue; }` for reducing one indentation.